[docs] Specify which props are added in the default `shouldForwardProp` option (#30978)
mui/material-ui: docs/data/system/styled/styled.md

1. It uses MUI's default `theme` if no theme is available in React context.
2. It supports the theme's [`styleOverrides`](/customization/theme-components/#global-style-overrides) and [`variants`](/customization/theme-components/#adding-new-component-variants) to be applied, based on the `name` applied in the options (can be skipped).
3. It adds support for the [the `sx` prop](/system/basics/#the-sx-prop) (can be skipped).

> 4. It adds by default `shouldForwardProp` option that is taking into account all props used internally in the MUI components: `ownerState`, `theme`, `sx`, `as` (can be overridden).

Reviewer: ```suggestion
4. It adds by default the `shouldForwardProp` option, taking into account all props used internally in the MUI components: `ownerState`, `theme`, `sx`, and `as` (which can be overridden).
```

> 4. It adds by default the `shouldForwardProp` option, taking into account all props used internally in the MUI components: `ownerState`, `theme`, `sx`, and `as` (which can be overridden).

Author: ```suggestion
4. It adds by default the `shouldForwardProp` option (that can be overridden), taking into account all props used internally in the MUI components: `ownerState`, `theme`, `sx`, and `as`.
```
